Assets/HotmaxEditorVR/activeProp.cs

```diff
@@ -0,0 +1,170 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class activeProp : MonoBehaviour
+{
+
+    //--local refs
+    private GameObject _selectedObject = stateManager.selectedObject;
+    private bool _selectedObjectIsActive = stateManager.selectedObjectIsActive;
+    private stateManager.editorModes _editorMode = stateManager.editorMode;
+
+    //--put the monobehaviours into a list
+    private List<MonoBehaviour> decorators = new List<MonoBehaviour>();
+
+    private void OnEnable()
+    {
+        stateManager.selectedObjectEvent += updateSelectedObject;
+        stateManager.editorModeEvent += updateEditorMode;
+        stateManager.selectedObjectIsActiveEvent += updateSelectedObjectIsActive;
+        addDecorations();
+    }
+
+    private void OnDisable()
+    {
+        stateManager.selectedObjectEvent -= updateSelectedObject;
+        stateManager.editorModeEvent -= updateEditorMode;
+        stateManager.selectedObjectIsActiveEvent -= updateSelectedObjectIsActive;
+        removeDecorators();
+    }
+
+    //this only fires when a new object is selected
```

Comment:
remove this comment (kind of obvious, as it is an event)

Reply:
But actually, leave a note that says `//destroy this component (and all decorators) when selected object changes`

Reply:
good point, got it